src/api/operations/form.queries.graphql

```diff
@@ -75,6 +75,53 @@ query GetFormDefinitions(
   }
 }
 
+query GetFormIdentifierForEditor($identifier: String!) {
+  formIdentifier(identifier: $identifier) {
```

Comment:
This feels like a _lot_ of data to resolve on page load for the page `/admin/forms/:identifier`. Can we re-work this so that the `FormDefinitionFieldsForEditor` fragment is only fetched when you're actually loading the Editor (or the Preview) for a specific form version? Some forms can be really big, there is no need to load all that json just to view the detail page. (here we are loading each json blob 2x or more per version).

Reply:
From further review, I see that we use the same query `useGetFormIdentifierForEditorQuery ` for both the "form details page" and the "edit form page", and that this issue predates your PR. That is worth changing imo, but you're welcome to do it in a separate PR if this one is feeling big.

An API that would make more sense to me is:
1. `GetFormIdentifierDetails($identifier: String!)` -- a query that loads the details about the form (by `identifier`) but not the form json itself. Used to back the form details page.
2. `GetFormDefinitionFieldsForEditor($id: ID!)` -- a query that loads the `FormDefinitionFieldsForEditor` fragment on a specific form version, by ID. Depending on the situation, this would either be the id of the published version or the draft versions.

For (2) I could see other approaches (mentioned in another comment). But (1) seems pretty standard to me.

Reply:
Agreed, updated to add the two queries you described.